components/dialog/demo/dialog-async.html

```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta http-equiv="X-UA-Compatible" content="ie=edge">
+		<meta charset="UTF-8">
+		<link rel="stylesheet" href="../../demo/styles.css" type="text/css">
+		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+		<script>
+			//window.D2L = { DialogMixin: { preferNative: false } };
+		</script>
+		<script type="module">
+			import '../../demo/demo-page.js';
+			import '../../button/button.js';
+			import '../dialog.js';
+			import './dialog-async-content.js';
+		</script>
+	</head>
+	<body unresolved>
+
+		<d2l-demo-page page-title="d2l-dialog">
+
+			<h2>Dialog (async)</h2>
+
+			<d2l-demo-snippet>
+				<template>
+					<d2l-button id="open">Show Dialog</d2l-button>
+					<d2l-dialog id="dialog" title-text="Dialog Title" async>
+						<d2l-dialog-demo-async-content></d2l-dialog-demo-async-content>
+						<d2l-button slot="footer" primary dialog-action="ok">Click Me!</d2l-button>
+						<d2l-button slot="footer" dialog-action>Cancel</d2l-button>
+					</d2l-dialog>
+					<script>
+						document.querySelector('#open').addEventListener('click', () => {
+							document.querySelector('#dialog').opened = true;
+							//document.querySelector('#dialog').resetAsyncState();
+							document.querySelector('d2l-dialog-demo-async-content').href = 'some-href';
```

Comment:
I don't quite get the purpose of the href property in this demo component...

Reply:
The runAsync directive used by the async component requires a key (typically it would be a property value), and when it changes, `render` gets rerun, and the async function get kicked off (returning a result after 5s using `setTimeout`). If I define this in advance or hard-code the key inside the component, the async state will be complete before the user opens the dialog - a valid scenario but it doesn't really demonstrate the state change.

So the purpose of the demo, I want to trigger the async function when the dialog opens so that we see the spinner and then the result.  I've tried to configure the demo component to match what I expect consumers to do (aside from the `setTimeout`).